lsp/document.go

```diff
@@ -0,0 +1,54 @@
+package lsp
+
+import (
+	"strings"
+
+	protocol "github.com/tliron/glsp/protocol_3_16"
+)
+
+type document struct {
+	uri     protocol.DocumentUri
+	Content string
+	lines   []string
+}
+
+type documentStore struct {
+	documents map[protocol.DocumentUri]*document
+}
+
+func newDocumentStore() *documentStore {
+	return &documentStore{
+		documents: make(map[protocol.DocumentUri]*document),
+	}
+}
+
+func (ds *documentStore) Get(uri protocol.DocumentUri) (*document, bool) {
+	doc, ok := ds.documents[uri]
+	return doc, ok
+}
+
+func (ds *documentStore) Set(uri protocol.DocumentUri, content string) {
+	ds.documents[uri] = &document{
+		uri:     uri,
+		Content: content,
+		lines:   strings.Split(content, "\n"),
```

Comment:
This looks like it will keep two copies of each file in RAM, is that a good idea?

Reply:
Maybe I'm over-optimizing here, but I was trying to reduce the amount of work the LSP server has to do on quick interactions like `hover` and instead refresh the "cache" when files are opened or changed.